cli/src/pcluster/models/imagebuilder.py

```diff
@@ -368,6 +387,52 @@ def _upload_artifacts(self):
                 f"Unable to upload imagebuilder cfn template to the S3 bucket {self.bucket.name} due to exception: {e}"
             )
 
+    def delete(self, force=False):
+        """Delete CFN Stack and associate resources and deregister the image."""
+        # validate image name
+        self._validate_image_name()
+
+        # TODO: check image is shared
+        if force or not self._check_instance_using_image():
+            try:
+                if AWSApi.instance().ec2.image_exists(self.image_name):
+                    # Deregister image
+                    AWSApi.instance().ec2.deregister_image(self.image.id)
+
+                    # Delete snapshot
+                    for snapshot_id in self.image.snapshot_ids:
+                        AWSApi.instance().ec2.delete_snapshot(snapshot_id)
+                elif AWSApi.instance().cfn.stack_exists(self.image_name):
+                    # Delete stack
+                    AWSApi.instance().cfn.delete_stack(self.image_name)
+
+                # Delete s3 image directory
+                try:
+                    self.bucket.check_bucket_exists()
+                    self.bucket.delete_s3_artifacts()
+                except AWSClientError:
+                    logging.warning("S3 bucket %s does not exist, skip image s3 artifacts deletion.", self.bucket.name)
+
+            except AWSClientError as e:
+                raise ImageBuilderActionError(f"Unable to delete image and stack, due to {str(e)}")
+
+    def _check_instance_using_image(self):
```

Comment:
Could you add a comment to this method?

Reply:
done

Reply:
What about `_is_image_in_use` or something like this?